react/features/file-sharing/middleware.web.ts

```diff
@@ -2,18 +2,44 @@ import { v4 as uuidv4 } from 'uuid';
 
 import { IStore } from '../app/types';
 import { getCurrentConference } from '../base/conference/functions';
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
 import { getLocalParticipant, getParticipantDisplayName } from '../base/participants/functions';
 import MiddlewareRegistry from '../base/redux/MiddlewareRegistry';
+import StateListenerRegistry from '../base/redux/StateListenerRegistry';
 import { showErrorNotification } from '../notifications/actions';
 import { NOTIFICATION_TIMEOUT_TYPE, NOTIFICATION_TYPE } from '../notifications/constants';
 
-import { DOWNLOAD_FILE, REMOVE_FILE, UPLOAD_FILES } from './actionTypes';
-import { addFile, removeFile, updateFileProgress } from './actions';
-import { FILE_SHARING_PREFIX, FILE_SHARING_SERVICE } from './constants';
+import { DOWNLOAD_FILE, FILE_LIST_RECEIVED, REMOVE_FILE, UPLOAD_FILES } from './actionTypes';
+import { addFile, fileRemoved, removeFile, updateFileProgress } from './actions';
 import { getFileExtension } from './functions.any';
 import logger from './logger';
 import { IFileMetadata } from './types';
 
+
+/**
+ * Registers a change handler for state['features/base/conference'].conference to
+ * set the event listeners needed for the file sharing feature to operate.
+ */
+StateListenerRegistry.register(
+    state => state['features/base/conference'].conference,
+    (conference, { dispatch }, previousConference) => {
+        if (conference && !previousConference) {
+            conference.on(JitsiConferenceEvents.FILE_SHARING_FILE_ADDED, (file: IFileMetadata) => {
+                dispatch(addFile(file));
+            });
+            conference.on(JitsiConferenceEvents.FILE_SHARING_FILE_REMOVED, (fileId: string) => {
+                dispatch(fileRemoved(fileId));
```

Comment:
Ah I see. I'd avoid having actions defined for these, just dispatch the object directly and use `_FOO` as the action name, since it's not an action originated from us, but rather a way for us to sync remote state.